knimin/lib/data_access.py

```diff
@@ -2588,110 +2615,168 @@ def _sample_exists(self, sample_id):
         bool
             Whether the sample exists
         """
-        sql = """SELECT EXISTS (SELECT 1 FROM pm.sample
-                                WHERE sample_id = %s)"""
-        return self._con.execute_fetchone(sql, [sample_id])[0]
+        with TRN:
+            sql = """SELECT EXISTS (SELECT 1 FROM pm.sample
+                                    WHERE sample_id = %s)"""
+            TRN.add(sql, [sample_id])
+            return TRN.execute_fetchlast()
 
-    def _barcode_exists(self, barcode):
-        """Confirms that a barcode exists
+    def _samples_exist(self, sample_ids, exist=True):
+        """Confirms that given sample ID(s) (do not) exist
 
         Parameters
         ----------
-        barcode : str
-            Barcode to check
+        sample_ids : list of str
+            IDs of the sample to check
+        exist : bool, optional, default: True
+            True: checks if exist
+            False: checks if not exist
+
+        Raises
+        ------
+        ValueError
+            If one or more sample IDs do not exist (when exist == True)
+            If one or more sample IDs already exist (when exist == False)
+        """
+        with TRN:
+            sql = """SELECT sample_id
+                     FROM pm.sample
+                     WHERE sample_id IN %s"""
+            TRN.add(sql, [tuple(sample_ids)])
+            res = [x[0] for x in TRN.execute_fetchindex()]
+            if exist:
+                res = set(sample_ids) - set(res)
+                if res:
+                    raise ValueError('Sample ID(s) %s do not exist.' %
+                                     ', '.join(sorted(res)))
+            else:
+                if res:
+                    raise ValueError('Sample ID(s) %s already exist.' %
+                                     ', '.join(sorted(res)))
+
+    def _validate_samples(self, samples):
+        """Confirms that the samples to be created or edited have valid
+        barcodes and study IDs
+
+        Parameters
+        ----------
+        samples : list of dict
+            {barcode : str, optional,
+                Assigns a barcode to the sample
```

Comment:
Update documentation, this function is not assigning/associating anything, is just validating.